inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/DynamicSchemaHandleOperator.java

```diff
@@ -252,7 +265,7 @@ private void handleDirtyData(Object dirtyData,
             dirtySinkHelper.invoke(dirtyData, dirtyType, dirtyOptions.getLabels(), dirtyOptions.getLogTag(),
                     dirtyOptions.getIdentifier(), e);
         }
-        if (metricData != null) {
+        if (metricData != null && needDirtyMetric) {
             metricData.outputDirtyMetricsWithEstimate(tableId.namespace().toString(), null, tableId.name(), dirtyData);
```

Comment:
Why not remove `metricData.outputDirtyMetricsWithEstimate(tableId.namespace().toString(), null, tableId.name(), dirtyData);`. DynamicSchemaHandleOperator pass dirty data metrics by RecordWithSchema.

Reply:
fixed